src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -594,11 +603,119 @@ def get_household_structure(pop: pd.DataFrame) -> pd.DataFrame:
         return household_structure
 
     @staticmethod
-    def choose_random_guardian(member_ids: pd.DataFrame) -> pd.Series:
+    def choose_random_guardian(member_ids: pd.DataFrame, groupby_level: str) -> pd.Series:
         # member_ids is a subset of child_households dataframe
+        # groupby_level will be index level to group by (the first level index of member_ids (child_id or mother_id).
         member_ids = (
             member_ids.reset_index()
-            .groupby(["child_id"])["person_id"]
+            .groupby([groupby_level])["person_id"]
             .apply(np.random.choice)
```

Comment:
Not part of this PR, but is `np.random.choice` appropriate or does it need to be a predictable randomness stream like we usually do?

Reply:
Since we're not running multiple scenarios the common random number (CRN) framework is less valuable. Also, since the point is to generate simulated data, the specifics don't really matter as long as they're internally consistent. 

The most important thing is to be able to harness numpy for performance reasons.